tool_wrapper.py

```diff
@@ -63,74 +23,77 @@ def setup_argparse():
     parser.add_argument('--tool')
     parser.add_argument('--env', default=[], action='append')
     parser.add_argument('--input')
-    parser.add_argument('--inputs')
     parser.add_argument('--tool-output')
     parser.add_argument('--output')
     parser.add_argument('--flag', default=[], action='append')
-    return parser
 
+    file_parser = argparse.ArgumentParser()
+    file_parser.add_argument('--file', default=[], action='append')
+    return file_parser, parser
 
-def main():
-    parser = setup_argparse()
-    args = parser.parse_args()
-    
-    tools = args.dep_tool + [args.tool]
 
-    env = dict(os.environ)
-    for en in args.env:
-        key, value = en.split(":")
-        value = os.path.abspath(value)
-        env[key] = value
-    texinputs = setup_dependencies()
-    # Add custom dependencies to TEXINPUTS
-    dependency_list = args.inputs.split(',')
-    texinputs.extend([os.path.abspath(path) for path in dependency_list])
-    env = setup_env(env, texinputs, tools)
-    cmd_args = [
-        os.path.basename(args.tool),
-    ] + args.flag + [args.input]
-
-    returncode = 0
-    output = None
-    engine = "luahbtex"
-    progname = "lualatex"
-
-    for flag in args.flag:
-        if "--latex-cmd=" in flag:
-            engine = flag[len("--latex-cmd="):]
-    ini_files = ["texmf", "texmf-dist", "tex", "generic", "tex-ini-files"]
-    ini_files_path = os.path.sep.join(ini_files)
-    engine_config_args = [engine, "-ini", "{}{}{}.ini".format(ini_files_path, os.path.sep, progname)]
-
-    try:
-        subprocess.check_output(
-            args=engine_config_args,
-            env=env,
-            stderr=subprocess.STDOUT,
-            universal_newlines=True,
-        )
-
-        subprocess.check_output(
-            args=cmd_args,
-            env=env,
-            stderr=subprocess.STDOUT,
-            universal_newlines=True,
-        )
-    except subprocess.CalledProcessError as exc:
-        output = exc.output
-        returncode = exc.returncode
-
-    if not os.path.exists(args.tool_output) or returncode != 0:
-        raise SystemExit(
-                """{} exited ({}) with:
+def main():
+    file_parser, parser = setup_argparse()
+    files = file_parser.parse_args().file
+    i = 0
+    for args_file in files:
+        with open(args_file) as fd:
+            args = parser.parse_args(fd.read().splitlines())
+        tools = args.dep_tool + [args.tool]
+
+        env = setup_env(args.env)
+        cmd_args = [args.tool] + args.flag + [args.input]
+        returncode = 0
+        output = None
+        res = ""
+        if not args.input:
+            cmd_args = cmd_args[:-1]
+
+        cwd="."
+        shell = False
```

Comment:
I'm not sure keeping `cwd` and `shell` declared independently from the call just below brings much value.

Reply:
No, this was probably a remains from debugging. Will fix.

Reply:
done